### DISCUSSION TITLE
"Cannot find module '...' or its corresponding type declarations."

### DISCUSSION THREAD

**Discussion by dcharbonnier** [panva/jose#141] [Q&A] [answered]
Can be more tricky than the answer in the FAQ :
```
/JwksManager.spec.ts:3:31 - error TS2307: Cannot find module 'jose' or its corresponding type declarations.
3 import { JWK as JWKLib } from "jose";
```
```
{
  "compilerOptions": {
    "target": "ES2020",
    "module": "commonjs",
    "sourceMap": true,
    "skipLibCheck": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "lib": ["es2020", "dom"],
    "typeRoots": ["./node_modules/@types", "./types"]
  }
}
```
```

    "typescript": "4.1.3"
    "jose": "3.5.0",
    "@types/node": "14.14.20",
node --version
v14.15.0
```

**Comment by panva** [comment]
Please refer to the docs. Your used syntax suggests you use the 2.x API but your package.json one is 3.x.

**Comment by dcharbonnier** [comment]
It's not a question of syntax, this syntax may be wrong but JWK still exists probably.
My ide do automatically the imports if I remove the import line (what I also did) and for my ide there is no JWK accessible anywhere.
jose is the only package with this problem

**Reply by panva** [reply]
> this syntax may be wrong but JWK still exists probably

Sigh. Probably? Do you think you probably use the module right? In v 3.x that your package json refers to there's no "root" jose export. And there's no exported member "JWK". Read the individual module docs. Or change your required package.json version to v2.x which uses the export and module structure your code shows.

**Comment by dcharbonnier** [comment]
Thanks, I've read it. Will stick to 2.x until it's fixed in 4.x, not a problem.

**Reply by panva** [reply]
We're not on the same page, there's nothing to "fix" in v3.x, you just need to follow its **docs** and understand how its individual modules are required/imported.

**Comment by dcharbonnier** [comment]
Don't worry I get it and I've read the documentation, that's why I'm gonna wait for 4.x

**Reply by panva** [reply]
Yeah. What for? No major bump is currently planned or needed.

**Comment by panva** [comment, marked as answer]
I'm done waiting for these ecosystems to catch up. jose v4.x will be coming out today/tomorrow with only top level named exports that solve all these tools' problems.

**Reply by panva** [reply]
See v4.x released.
